native/App.tsx

```diff
@@ -122,15 +118,21 @@ const App = () => {
   const [theme, setTheme] = useState<'dark' | 'light'>('light')
 
   const setEndpointWithNativeQuery = (endpoint?: string) => {
-    const newUrl = baseUri + (endpoint ?? 'home') + nativeQuery
+    const newUrl =
+      baseUri +
+      (endpoint ?? 'home') +
+      nativeQuery +
+      `&rand=${Math.random().toString()}`
```

Comment:
Adding a random string ensures the `<WebView/>` `src` parameter will treat a new url from our `setUrlToLoad` as a new state even if it was previously set to the same url.